src/rpc_request.rs

```diff
@@ -6,6 +6,7 @@ use std::thread::sleep;
 use std::time::Duration;
 use std::{error, fmt};
 
+#[derive(Clone)]
 pub struct RpcClient {
```

Comment:
Instead of `Rpu` and `MockRpu`, would it make more sense to implement your trait for this struct?  So there'd be a `RpcClient` and `MockRpcClient`?  The your code would read `rpc_client.make_rpc_request(...)` and `rpc_client` would be either an instance of `RcpClient` or `MockRpcClient`.  Currently, it's `Rpu::make_rpc_request(&client, ...)`. Thus, no function that uses the explicit `Rpu` could be tested with `MockRpu`.

As it relates to my previous question asking if it makes sense to reintroduce the term `Rpu`, I suspect the correct answer was, "no @garious, that's a terrible suggestion. There's already a wrapper called RpcClient. It hosts the reqwest client, which wouldn't be used by a new `MockRpcClient`. Everyone in this forum is now dumber for having looked at your code review."

Reply:
Makes a lot more sense! Edits made. (I meant to reference Billy Madison in my commit message... Opportunity missed!)